src/core/project/qgsproject.cpp

```diff
@@ -1575,11 +1575,20 @@ bool QgsProject::readProjectFile( const QString &filename, QgsProject::ReadFlags
   }
   emit metadataChanged();
 
-  element = doc->documentElement().firstChildElement( QStringLiteral( "autotransaction" ) );
-  if ( ! element.isNull() )
+  // Transaction mode
+  element = doc->documentElement().firstChildElement( QStringLiteral( "transaction-mode" ) );
+  if ( !element.isNull() )
   {
-    if ( element.attribute( QStringLiteral( "active" ), QStringLiteral( "0" ) ).toInt() == 1 )
-      mAutoTransaction = true;
+    mTransactionMode = static_cast<Qgis::TransactionMode>( element.attribute( QStringLiteral( "active" ), QStringLiteral( "0" ) ).toInt() );
```

Comment:
```suggestion
    mTransactionMode = qgsEnumKeyToValue<Qgis::TransactionMode>( element.attribute( QStringLiteral( "mode" ), QSgring() ), Qgis::TransactionMode::None );
```

Reply:
Thanks, no need to explicitly specify the template `<Qgis::TransactionMode>` as the compiler can deduce it from the default value